src/audio/tdfb/tdfb.c

```diff
@@ -222,19 +337,25 @@ static int tdfb_setup(struct tdfb_comp_data *cd, int source_nch, int sink_nch)
 	if (!delay_size)
 		return 0;
 
-	/* Allocate all FIR channels data in a big chunk and clear it */
-	cd->fir_delay = rballoc(0, SOF_MEM_CAPS_RAM, delay_size);
-	if (!cd->fir_delay) {
-		comp_cl_err(&comp_tdfb, "tdfb_setup(), delay allocation failed for size %d",
-			    delay_size);
-		return -ENOMEM;
-	}
+	if (delay_size > cd->fir_delay_size) {
+		/* Free existing FIR channels data if it was allocated */
+		tdfb_free_delaylines(cd);
+
+		/* Allocate all FIR channels data in a big chunk and clear it */
+		cd->fir_delay = rballoc(0, SOF_MEM_CAPS_RAM, delay_size);
```

Comment:
Is this allocated once at init() or at multiple times ? i.e. do we need to track usage.

Reply:
There's logic in tdfb_setup() to use the existing allocation if the new arrived configuration fits the space. Is that OK when using rballoc()? I think the usage is such that re-configuration happens only in device tuning phase, or alternatively the system boots with a small pass-trough blob and some audio server later loads the actual blob.

Reply:
as long as this is tracked and freed().